### PR TITLE
Add pre-commit execution option

### PR DESCRIPTION
**Pull request by lorman**
Added pre-commit-hooks.yaml to enable inclusion as a pre-commit hook for https://pre-commit.com/

### CODE REVIEW

**Comment by adithyabsk** [comment]
I thought I'd bump this PR. This would be a great addition to the pre-commit arsenal

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-   id: rstcheck
+    name: rstcheck
+    entry: rstcheck
+    language: python
+    'types': [rst]
+    args: []
+    require_serial: false
+    additional_dependencies: []
```
Comment (asottile) [inline .pre-commit-hooks.yaml:8]: these three lines are unnecessary

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+-   id: rstcheck
+    name: rstcheck
+    entry: rstcheck
+    language: python
+    'types': [rst]
```
Comment (asottile) [inline .pre-commit-hooks.yaml:5]:
```suggestion
    types: [rst]
```

File: README.rst [hunk 3]
```diff
@@ -256,6 +256,17 @@ is the line number (not the line index). The second value is the error message.
 Note that this does not load any configuration as that would mutate the
 ``docutils`` registries.
 
+Use as a pre-commit hook
+========================
+
+Add this to your ``.pre-commit-config.yaml``
+
+.. code-block:: cfg
```
Comment (asottile) [inline README.rst:264]: I believe this should be `.. code-block:: yaml`

**Comment by lorman** [comment]
@asottile made the adjustments, thanks for the review.

**Pull request merged**